subprojects/plugins/src/main/java/org/gradle/api/plugins/internal/DefaultJavaFeatureSpec.java

```diff
@@ -152,16 +151,11 @@ private void setupConfigurations(SourceSet sourceSet) {
             configurationContainer.getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME).extendsFrom(impl);
         }
 
-        pluginManager.withPlugin("maven-publish", new Action<AppliedPlugin>() {
-            @Override
-            public void execute(AppliedPlugin plugin) {
-                final AdhocComponentWithVariants component = findComponent();
-                if (component != null) {
-                    component.addVariantsFromConfiguration(apiElements, new JavaConfigurationVariantMapping("compile", true));
-                    component.addVariantsFromConfiguration(runtimeElements, new JavaConfigurationVariantMapping("runtime", true));
-                }
-            }
-        });
+        final AdhocComponentWithVariants component = findComponent();
```

Comment:
I think you need to keep the `withPlugin` or it will cause ordering issues, depending on what plugin is applied first.

Reply:
Then it has a dependency on the `java` plugin, not a publication one. Since that's where the `java` component is added.

Reply:
Which ordering issue are you thinking of?
- If the publish plugin is applied first, this is done later anyway.
- If the publish plugin is applied after, I don't see what the problem can be. The _java_ component is from the Java plugin itself. And most tests apply the publishing plugin second and everything still works.
Or is there anything I am not seeing the publishing plugin could "overwrite"?

Reply:
mmm, no you're right, we should be safe.